cmd/terramate/cli/script_run.go

```diff
@@ -109,13 +111,31 @@ func (c *cli) runScript() {
 		}
 	}
 
-	c.prepareScriptCloudDeploymentSync(runs)
+	orderedStacks, reason, err := runutil.Sort(c.cfg(), stacks)
+	if err != nil {
+		if errors.IsKind(err, dag.ErrCycleDetected) {
+			fatal(err, "cycle detected: %s", reason)
+		} else {
+			fatal(err, "failed to plan execution")
+		}
+	}
+
+	var orderedRuns []runContext
+	for _, st := range orderedStacks {
+		for _, r := range runs {
+			if r.Stack.Dir.String() == st.Dir().String() {
+				orderedRuns = append(orderedRuns, r)
+			}
+		}
+	}
```

Comment:
note: some entries in the `orderedStacks` could be skipped and be not present in the final `orderedRuns`. That happens when there's no script definition for the stack, then we can't run them.